### ISSUE TITLE
bug: several queries result mismatch under parallel e2e

### ISSUE THREAD

**Issue by BugenZhao** [risingwavelabs/risingwave#4857]
https://buildkite.com/singularity-data/pull-request/builds/6736#0182cf11-3740-4cc4-b3a1-bb69a186aa9b

```
\| e2e_test/streaming/./tpch/q19.slt.part                       .. [FAILED]
--
  |  
  | failed to run `e2e_test/streaming/tpch_snapshot.slt`
  |  
  | Caused by:
  | test error at e2e_test/streaming/./tpch/q19.slt.part:1: query result mismatch:
  | [SQL] select revenue from tpch_q19;
  | [Diff]
  | 33530.5144
```

TPCH q19 is a simple aggregation, but we scan 0 records from the storage.

**Comment by BugenZhao** [comment]
Q18 also fails:

https://buildkite.com/singularity-data/pull-request/builds/6762#0182cf7f-8802-4e4a-9262-aee8cd8cb731

**Comment by xxchan** [comment]
https://buildkite.com/singularity-data/pull-request/builds/6772#0182d001-2ef9-4f09-80d6-d5f6510621a7

nexmark snapshot q4 & mv on mv 🤪

<img width="645" alt="image" src="https://user-images.githubusercontent.com/37948597/186438195-ef34fd26-f4ec-40de-a5d1-eb776852d5d7.png">

**Comment by BugenZhao** [comment]
Looks like a serious problem. 🥵🥵

**Comment by kwannoel** [comment]
q1 fails too: https://buildkite.com/singularity-data/pull-request/builds/6800#0182d2a1-b5bd-4dd4-b29c-183cdfd3d4f8

![2022-08-25-095808_706x287_scrot](https://user-images.githubusercontent.com/47273164/186556849-0a595a97-36b6-46d4-8fb3-340b77c8524d.png)

**Comment by wangrunji0408** [comment]
Q14 fails: https://buildkite.com/singularity-data/pull-request/builds/6802#0182d2d9-23d2-4a6e-9062-c750f95d9a27/205-475 🥵

<img width="646" alt="WX20220825-111953@2x" src="https://user-images.githubusercontent.com/15158738/186566903-2dfdc413-469b-4433-b38a-9580e36f59f5.png">

**Comment by hzxa21** [comment]
Looks all relevant to #4813.

**Comment by xxhZs** [comment]
> Looks all relevant to #4813.

https://github.com/singularity-data/risingwave/pull/4813 Can't solve the problem😭

**Comment by xxhZs** [comment]
relevant  https://github.com/singularity-data/risingwave/pull/4890

**Comment by lmatz** [comment]
Shall we let the CI run more rounds of the tests for highly critical/complicated PRs?

i.e., We associate a PR with a GitHub label `critical`. Buildkite figures out the label, and then test multiple rounds/spend more time in CI.

**Comment by xxhZs** [comment]
After https://github.com/singularity-data/risingwave/pull/4890 , I do some test under parallel e2e, It has not been reproduced again. If everyone is not reproducing again. Maybe we can close this issue.

**Comment by lmatz** [comment]
> After #4890 , I do some test under parallel e2e, It has not been reproduced again. If everyone is not reproducing again. Maybe we can close this issue.

It has been 6 days and about 3 1/2 pages of PRs have been merged since #4890,
guess we can conclude that the bug is fixed now? 🤔

**Comment by xxhZs** [comment]
It has been 8 days didn't see it happen again since https://github.com/risingwavelabs/risingwave/pull/4890, So close it
